src/redisearch_rs/rqe_iterators/tests/integration/inverted_index/term.rs

```diff
@@ -0,0 +1,143 @@
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use ffi::{
+    IndexFlags_Index_StoreByteOffsets, IndexFlags_Index_StoreFieldFlags,
+    IndexFlags_Index_StoreFreqs, IndexFlags_Index_StoreTermOffsets, t_docId, t_fieldMask,
+};
+use inverted_index::{RSIndexResult, RSOffsetVector, full::Full};
+use rqe_iterators::inverted_index::TermFull;
+
+use crate::{
+    ffi::query_term::QueryTermBuilder,
+    inverted_index::utils::{BaseTest, RevalidateTest},
+};
+
+struct TermTest {
+    test: BaseTest<Full>,
+    revalidate_test: RevalidateTest<Full>,
+}
+
+impl TermTest {
+    // # Safety
+    // The returned RSIndexResult contains raw pointers to `term` and `offsets`.
+    // These pointers are valid for 'static because the data is moved into the closure
+    // in `new()` and lives for the entire duration of the test. The raw pointers are
+    // only used within the test's lifetime, making this safe despite the 'static claim.
+    fn expected_record(
+        doc_id: t_docId,
+        term: *mut ffi::RSQueryTerm,
+        offsets: &Vec<u8>,
+    ) -> RSIndexResult<'static> {
+        RSIndexResult::term_with_term_ptr(
+            term,
+            RSOffsetVector::with_data(offsets.as_ptr() as _, offsets.len() as _),
+            doc_id,
+            (doc_id / 2) as t_fieldMask + 1,
+            (doc_id / 2) as u32 + 1,
+        )
+    }
+
+    fn new(n_docs: u64) -> Self {
+        let flags = IndexFlags_Index_StoreFreqs
+            | IndexFlags_Index_StoreTermOffsets
+            | IndexFlags_Index_StoreFieldFlags
+            | IndexFlags_Index_StoreByteOffsets;
+
+        const TEST_STR: &str = "term";
+
+        let offsets = vec![0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+        let offsets_clone = offsets.clone();
+
+        Self {
+            test: BaseTest::new(
+                flags,
+                Box::new(move |doc_id| {
+                    let term = QueryTermBuilder {
+                        token: TEST_STR,
+                        idf: 5.0,
+                        id: 1,
+                        flags: 0,
+                        bm25_idf: 10.0,
+                    }
+                    .allocate();
+                    Self::expected_record(doc_id, term, &offsets)
+                }),
+                n_docs,
+            ),
+            revalidate_test: RevalidateTest::new(
+                IndexFlags_Index_StoreTermOffsets,
+                Box::new(move |doc_id| {
+                    let term2 = QueryTermBuilder {
+                        token: TEST_STR,
+                        idf: 5.0,
+                        id: 1,
+                        flags: 0,
+                        bm25_idf: 10.0,
+                    }
+                    .allocate();
+                    Self::expected_record(doc_id, term2, &offsets_clone)
+                }),
+                n_docs,
+            ),
+        }
+    }
+}
+
+#[test]
+/// test reading from TermFull iterator
+fn term_full_read() {
```

Comment:
Didn't we drop these implementations? "Full" variants are redundant in Rust as we have a reader over the index (CC @gdesmott)